File: src/main/java/ru/iamdvz/wit/functions/WITFunc.java
package ru.iamdvz.wit.functions;

import org.bukkit.configuration.ConfigurationSection;
import org.bukkit.entity.Entity;
import ru.iamdvz.wit.utils.MythicMobs;
import ru.iamdvz.wit.utils.Utils;

import java.util.Objects;

public class WITFunc {

    private static String stringFormat = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("format-of-mythic-mobs-string");
    private static boolean showLvl = ru.iamdvz.wit.WIT.getInstance().getConfig().getBoolean("show-level-of-mythic-mobs");
    private static boolean showHealth = ru.iamdvz.wit.WIT.getInstance().getConfig().getBoolean("show-health-of-mythic-mobs");
    private static String emptyChar = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("mythic-mob-health-bar-empty");
    private static String fullChar = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("mythic-mob-health-bar-full");
    public static String nothingString = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("nothing-string");
    private static ConfigurationSection mobAndItString = ru.iamdvz.wit.WIT.getInstance().getConfig().getConfigurationSection("mob-and-it-parameters");
    public static double checkerDistance = ru.iamdvz.wit.WIT.getInstance().getConfig().getDouble("checker-distance");

    public static void WITPlaceholdersReload() {
        stringFormat = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("format-of-mythic-mobs-string");
        showLvl = ru.iamdvz.wit.WIT.getInstance().getConfig().getBoolean("show-level-of-mythic-mobs");
        showHealth = ru.iamdvz.wit.WIT.getInstance().getConfig().getBoolean("show-health-of-mythic-mobs");
        emptyChar = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("mythic-mob-health-bar-empty");
        fullChar = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("mythic-mob-health-bar-full");
        nothingString = ru.iamdvz.wit.WIT.getInstance().getConfig().getString("nothing-string");
        mobAndItString = ru.iamdvz.wit.WIT.getInstance().getConfig().getConfigurationSection("mob-and-it-parameters");
        checkerDistance = ru.iamdvz.wit.WIT.getInstance().getConfig().getDouble("checker-distance");
    }

    public static String outputFunction(Entity entity) {
        if (!MythicMobs.isMythicMob(entity)) {return nothingString;}
        String s = stringFormat;
        s = s.replace("{name}", Objects.requireNonNull(MythicMobs.getMythicMobDisplayName(entity)));
        if (showLvl) s = s.replace("{level}", String.valueOf(MythicMobs.getMythicMobLevel(entity)));
        if (showHealth) s = s.replace("{health}", String.valueOf(MythicMobs.getMythicMobCurrentHealth(entity)));
        return s.replace("{level}", "").replace("{health}", "");
    }

    public static String healthBarFunc(Entity entity, int size) {
        if (!MythicMobs.isMythicMob(entity)) {return nothingString;}
        return Utils.makeBar(emptyChar, fullChar, size,
                MythicMobs.getMythicMobMaxHealth(entity), MythicMobs.getMythicMobCurrentHealth(entity));
    }
    public static String getParameter(Entity entity, String parameter) {
        if (!MythicMobs.isMythicMob(entity)) {return nothingString;}
        if (mobAndItString.contains(MythicMobs.getMythicMobConfigName(entity))) {
            return mobAndItString.getString(MythicMobs.getMythicMobConfigName(entity) + "." + parameter);
        }
        return nothingString;
    }
}

File: src/main/java/ru/iamdvz/wit/functions/WITFunc.kt
package ru.iamdvz.wit.functions

import org.bukkit.entity.Entity
import ru.iamdvz.wit.WIT
import ru.iamdvz.wit.utils.MythicMobUtil
import ru.iamdvz.wit.utils.Utils.makeBar
import java.util.*

object WITFunc {
    private var stringFormat = WIT.getInstance().config.getString("format-of-mythic-mobs-string")!!
    private var showLvl = WIT.getInstance().config.getBoolean("show-level-of-mythic-mobs")
    private var showHealth = WIT.getInstance().config.getBoolean("show-health-of-mythic-mobs")
    private var emptyChar = WIT.getInstance().config.getString("mythic-mob-health-bar-empty")!!
    private var fullChar = WIT.getInstance().config.getString("mythic-mob-health-bar-full")!!
    var nothingString = WIT.getInstance().config.getString("nothing-string")!!
    private var mobAndItString = WIT.getInstance().config.getConfigurationSection("mob-and-it-parameters")!!
    var checkerDistance = WIT.getInstance().config.getDouble("checker-distance")

    @JvmStatic
    fun WITReload() {
        stringFormat = WIT.getInstance().config.getString("format-of-mythic-mobs-string")!!
        showLvl = WIT.getInstance().config.getBoolean("show-level-of-mythic-mobs")
        showHealth = WIT.getInstance().config.getBoolean("show-health-of-mythic-mobs")
        emptyChar = WIT.getInstance().config.getString("mythic-mob-health-bar-empty")!!
        fullChar = WIT.getInstance().config.getString("mythic-mob-health-bar-full")!!
        nothingString = WIT.getInstance().config.getString("nothing-string")!!
        mobAndItString = WIT.getInstance().config.getConfigurationSection("mob-and-it-parameters")!!
        checkerDistance = WIT.getInstance().config.getDouble("checker-distance")
    }

    @JvmStatic
    fun outputFunction(entity: Entity?): String {
        if (!MythicMobUtil.isMythicMob(entity)) {
            return nothingString
        }
        var s = MythicMobUtil.getMythicMobDisplayName(entity)?.let { stringFormat.replace("{name}", it) }
        if (showLvl) s = s?.replace("{level}", MythicMobUtil.getMythicMobLevel(entity).toString())
        if (showHealth) s = s?.replace("{health}", MythicMobUtil.getMythicMobCurrentHealth(entity).toString())
        return s?.replace("{level}", "")?.replace("{health}", "") ?: "NULL"
    }

    @JvmStatic
    fun healthBarFunc(entity: Entity?, size: Int): String {
        return if (!MythicMobUtil.isMythicMob(entity)) { nothingString
        } else makeBar(emptyChar, fullChar, size, MythicMobUtil.getMythicMobMaxHealth(entity), MythicMobUtil.getMythicMobCurrentHealth(entity))
    }

    @JvmStatic
    fun getParameter(entity: Entity?, parameter: String): String? {
        if (!MythicMobUtil.isMythicMob(entity)) {
            return nothingString
        }
        return if (mobAndItString.contains(MythicMobUtil.getMythicMobConfigName(entity))) {
            mobAndItString.getString(MythicMobUtil.getMythicMobConfigName(entity) + "." + parameter)
        } else nothingString
    }
}